ephios/static/ephios/js/serviceworker.js

```diff
@@ -19,40 +8,70 @@ const filesToCache = [
     "/static/ephios/img/ephios-text-black.png",
 ];
 
-// Cache on install
 self.addEventListener("install", event => {
-    this.skipWaiting();
     event.waitUntil(
-        caches.open(staticCacheName)
-            .then(cache => {
-                return cache.addAll(filesToCache);
-            })
-    )
+        caches.open(staticCacheName).then(cache => {
+            return cache.addAll(staticFilesToCacheOnInstall);
+        })
+    );
 });
 
-// Clear cache on activate
 self.addEventListener('activate', event => {
     event.waitUntil(
         caches.keys().then(cacheNames => {
             return Promise.all(
                 cacheNames
-                    .filter(cacheName => (cacheName.startsWith("django-pwa-")))
+                    .filter(cacheName => (cacheName.startsWith("ephios-pwa-")))
                     .filter(cacheName => (cacheName !== staticCacheName))
                     .map(cacheName => caches.delete(cacheName))
             );
         })
     );
 });
 
-// Serve from Cache
 self.addEventListener("fetch", event => {
     event.respondWith(
-        caches.match(event.request)
-            .then(response => {
-                return response || fetch(event.request);
-            })
-            .catch(() => {
-                return caches.match('/offline/');
-            })
```

Comment:
Problem might actually be in the cache management. See how here I'm looking at *all* caches while in the new version only in the cache newly selected.